Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/approximated_centroidal_Voronoi_diagram_remeshing.h

```diff
@@ -0,0 +1,1319 @@
+*     \cgalParamType{`GT::FT`}
+*     \cgalParamDefault{0}
+*     \cgalParamExtra{If this parameter is omitted, no adaptive clustering will be performed.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{use_postprocessing_qem}
+*     \cgalParamDescription{indicates if a projection step using quadric error metric should be applied to cluster centers at the end of the minimization,
+*                           in order for example to recover sharp features.
+*                           This is a fast method but can result in badly shaped triangles and even self-intersections.}
+*     \cgalParamType{bool}
+*     \cgalParamDefault{false}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{use_qem_based_energy}
+*     \cgalParamDescription{indicates if quadric error metric should be applied during the minimization algorithm in order for example to recover sharp features.
+*                           This is slower than when using `use_postprocessing_qem(true)` but it is more accurate.}
+*     \cgalParamType{bool}
+*     \cgalParamDefault{false}
+*     \cgalParamExtra{If this parameter is `true` then `use_postprocessing_qem` will automatically set to `false`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_count_ratio}
+*     \cgalParamDescription{a ratio used to control the subdivision of the input mesh in case it does not have enough vertices compared to `nb_vertices`.
+*                           More precisely, the number of vertices of the input mesh should at least be the ratio times `nb_vertices`.
+*                           If not it will be internally subdivided until the aforementioned criterium is met.}
+*     \cgalParamType{`GT::FT`}
+*     \cgalParamDefault{0.1}
+*     \cgalParamExtra{A value between 0.1 and 0.01 is recommended, the smaller the better the approximation will be, but it .}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_point_map}
+*       \cgalParamDescription{a property map associating points to the vertices of `tmesh`.}
+*       \cgalParamType{a class model of `ReadablePropertyMap` with
+*                      `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+*                      as key type and `GT::Point_3` as value type.}
+*       \cgalParamDefault{`boost::get(CGAL::vertex_point, tmesh)`.}
+*       \cgalParamExtra{If this parameter is omitted, an internal property map for
+*                       `CGAL::vertex_point_t` must be available in `TriangleMesh`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{geom_traits}
+*      \cgalParamDescription{an instance of a geometric traits class.}
+*      \cgalParamType{a class model of `Kernel`}
+*      \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`.}
+*      \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+*   \cgalParamNEnd
+*
+* \cgalNamedParamsEnd
+*
+* @return `true` if `nb_vertices` was sufficiently large for remeshing the input, and `false` if more points were used
```

Comment:
That's not what is returned: `return ps.first.size()==nb_vertices;` so you're checking if the final point set has the user-desired value.